source/dagger/gameplay/team_game/door_interaction.cpp

```diff
@@ -15,52 +15,58 @@ using namespace team_game;
 
 using namespace dagger;
 
-bool keytaken = false;
 
+Bool pickedup = false;
```

Comment:
can't u put it inside DoorSystem at least? as a member field

Reply:
Thank you for all of the nice suggestions 😃